Components/Queues/Handler/PhpAmqpHandler.php

```diff
@@ -0,0 +1,232 @@
+use Smartbox\Integration\FrameworkBundle\Components\Queues\QueueMessageInterface;
+use Smartbox\Integration\FrameworkBundle\Core\Consumers\IsStopableConsumer;
+use Smartbox\Integration\FrameworkBundle\Core\Endpoints\EndpointFactory;
+use Smartbox\Integration\FrameworkBundle\Core\Endpoints\EndpointInterface;
+use Smartbox\Integration\FrameworkBundle\Core\Messages\MessageInterface;
+use Smartbox\Integration\FrameworkBundle\DependencyInjection\Traits\UsesSmartesbHelper;
+use Smartbox\Integration\FrameworkBundle\Exceptions\Handler\UsesExceptionHandlerTrait;
+
+/**
+ * Class PhpAmqpHandler
+ * @package Smartbox\Integration\FrameworkBundle\Components\Queues\Handler
+ */
+class PhpAmqpHandler implements LoggerAwareInterface
+{
+    use LoggerAwareTrait;
+    use UsesSmartesbHelper;
+    use UsesExceptionHandlerTrait;
+    use IsStopableConsumer;
+
+    /**
+     * @var EndpointInterface
+     */
+    private $endpoint;
+
+    /**
+     * Format used by serializer function
+     * @var string
+     */
+    private $format;
+
+    /**
+     * @var SerializerInterface
+     */
+    private $serializer;
+
+    /**
+     * @var AMQPChannel
+     */
+    private $channel;
+
+    /**
+     * @param EndpointInterface        $endpoint   The endpoint used to consume
+     * @param int                      $expirationCount  Maximum amount of message to consume before stopping
+     * @param string                   $format     json|xml|php
+     * @param null $serializer Serializer to use for deserialization
+     */
+    public function __construct(EndpointInterface $endpoint, int $expirationCount = -1, string $format = QueueDriverInterface::FORMAT_JSON, $serializer = null)
+    {
+        $this->endpoint = $endpoint;
+        $this->setExpirationCount($expirationCount);
+        $this->format = $format;
+        $this->serializer = $serializer ?? SerializerBuilder::create()->build();
+    }
+
+    /**
+     * Consumes the message and dispatch it to others features
+     * @param string $consumerTag
+     * @param AMQPChannel $channel
+     * @param string $queueName
+     * @throws \Exception
+     */
+    public function consume(string $consumerTag, AMQPChannel $channel, string $queueName)
+    {
+        $callback = function($message) use ($channel) {
+            $this->isConnected($channel);
+            // Send a message with the string "quit" to cancel the consumer.
+            if ($message->body === 'quit' || $this->shouldStop()) {
+                $channel->basic_cancel($message->delivery_info['consumer_tag']);
+                return false;
+            }
+
+            $this->log('A message was received on {time}');
+            $this->log('Message Body:' . $message->body);
+
+            $queueMessage = $this->deserializeMessage($message);
+            $this->dispatchMessage($queueMessage);
+            $channel->basic_ack($message->delivery_info['delivery_tag']);
```

Comment:
I get that you're using the callback approach and calling the internal functions of the library directly, but basically you're bypassing the definition of [our own interface](https://github.com/smartboxgroup/integration-framework-bundle/blob/master/Components/Queues/Drivers/QueueDriverInterface.php).

The handler should implement all the functions that the interface defines and this callback should not bypass them.

Reply:
This approach will be implemented on the future, maybe. The first idea was just to validate the driver and the features that we can do with it.